crates/bevy_plugin/src/dialogue_runner/runtime_interaction.rs

```diff
@@ -88,55 +96,107 @@ fn continue_runtime(
+            ),
+        );
+    }
 
-        for event in events {
-            match event {
-                DialogueEvent::Line(line) => {
-                    let assets = dialogue_runner.get_assets(&line);
-                    let metadata = project.line_metadata(&line.id).unwrap_or_default().to_vec();
-                    present_line_events.write(PresentLineEvent {
-                        line: LocalizedLine::from_yarn_line(line, assets, metadata),
-                        source,
-                    });
-                }
-                DialogueEvent::Options(options) => {
-                    let options: Vec<DialogueOption> = options
-                        .into_iter()
-                        .map(|option| {
-                            let assets = dialogue_runner.get_assets(&option.line);
-                            let metadata = project
-                                .line_metadata(&option.line.id)
-                                .unwrap_or_default()
-                                .to_vec();
-                            DialogueOption::from_yarn_dialogue_option(option, assets, metadata)
-                        })
-                        .collect();
-                    last_options.insert(source, options.clone());
-                    present_options_events.write(PresentOptionsEvent { options, source });
-                }
-                DialogueEvent::Command(command) => {
-                    execute_command_events.write(ExecuteCommandEvent { command, source });
-                    dialogue_runner.continue_in_next_update();
-                }
-                DialogueEvent::NodeComplete(node_name) => {
-                    node_complete_events.write(NodeCompleteEvent { node_name, source });
-                }
-                DialogueEvent::NodeStart(node_name) => {
-                    node_start_events.write(NodeStartEvent { node_name, source });
-                }
-                DialogueEvent::LineHints(line_ids) => {
-                    line_hints_events.write(LineHintsEvent { line_ids, source });
-                }
-                DialogueEvent::DialogueComplete => {
-                    if !is_sending_missed_events {
-                        dialogue_runner.is_running = false;
+    for (dialogue, _, unsent_events, events) in dialogues.values_mut() {
+        let new_events = if let Some(unsent_events) = unsent_events.take() {
+            unsent_events
+        } else {
+            dialogue.continue_(world)?
+        };
+        events.replace(new_events);
+    }
+
+    let mut system_state: SystemState<(
+        Query<(Entity, &mut DialogueRunner)>,
+        EventWriter<PresentLineEvent>,
+        EventWriter<PresentOptionsEvent>,
+        EventWriter<ExecuteCommandEvent>,
+        EventWriter<NodeCompleteEvent>,
+        EventWriter<NodeStartEvent>,
+        EventWriter<LineHintsEvent>,
+        EventWriter<DialogueCompleteEvent>,
+        Local<HashMap<Entity, Vec<DialogueOption>>>,
```

Comment:
`Local` would prevent us from running multiple dialogue runners, e.g. one per player, wouldn't it?